test/export/test_export.py

```diff
@@ -1040,6 +1040,41 @@ def forward(self, x):
         with self.assertRaisesRegex(NotImplementedError, r"Calling eval\(\) is not supported yet."):
             graph_module.eval()
 
+    def test_export_cond_preserve_stack_trace_for_subgraphs(self):
+        class MySubModule(torch.nn.Module):
+            def foo(self, x):
+                return x.cos()
+
+            def forward(self, x):
+                return self.foo(x)
+
+        class CondBranchClassMethod(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.subm = MySubModule()
+
+            def bar(self, x):
+                return x.sin()
+
+            def forward(self, x):
+                return cond(x.shape[0] <= 2, self.subm.forward, self.bar, [x])
+
+
+        from torch._export import capture_pre_autograd_graph
+
+        example_inputs = (torch.randn(1, 3, 3, 3),)
+        m = CondBranchClassMethod()
+        m.eval()
+        gm = capture_pre_autograd_graph(m, example_inputs)
+
+        actual_source_fns = []
+        for mod in gm.modules():
+            for node in mod.graph.nodes:
+                if node.name in {"sin", "cos"}:
+                    actual_source_fns.append(node.meta.get("source_fn", None))
+        exp_source_fns = [("cos", "cos"), ("sin", "sin")]
```

Comment:
I thought we'll have a stack of sources like ("cond", "cos")?

Reply:
right now, the original source_fn has this format (rv.node.name, target). See [here](https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/output_graph.py#L1281), it is a tuple ofnode name and target.

Righit now this pr just preserves the source_fn. Extending it can be next pr. It requires some changes in dynamo to add the "cond" to the source_fn. Do you think the "cond" is a necessary feature?